libs/wallet/src/web3-react/hooks/useSafeAppsSdkInfo.ts

```diff
@@ -1,22 +1,24 @@
-import { useEffect, useState } from 'react'
-
 import type { SafeInfo } from '@safe-global/safe-apps-sdk'
 
-import { useSafeAppsSdk } from './useSafeAppsSdk'
+import { useAtom } from 'jotai/index'
+import { gnosisSafeInfoAtom } from '@cowprotocol/wallet'
 
 export type GnosisSafeSdkInfo = SafeInfo
 
 export function useSafeAppsSdkInfo(): GnosisSafeSdkInfo | null {
-  const [gnosisSafeInfo, setGnosisSafeInfo] = useState<GnosisSafeSdkInfo | null>(null)
-  const safeAppsSdk = useSafeAppsSdk()
+  const gnosisSafeInfo = useAtom(gnosisSafeInfoAtom)[0]
+
+  console.log('main use safe apps sdk info')
 
-  useEffect(() => {
-    if (!safeAppsSdk) {
-      setGnosisSafeInfo(null)
-    } else {
-      safeAppsSdk.safe.getInfo().then(setGnosisSafeInfo)
-    }
-  }, [safeAppsSdk])
+  if (!gnosisSafeInfo) {
+    return null
+  }
 
-  return gnosisSafeInfo
+  return {
+    safeAddress: gnosisSafeInfo.address,
+    chainId: gnosisSafeInfo.chainId,
+    threshold: gnosisSafeInfo.threshold,
+    owners: gnosisSafeInfo.owners,
+    isReadOnly: !!gnosisSafeInfo.isReadOnly,
```

Comment:
nit: destructure

Reply:
my comment is not relevant, cause I think we should implement this differently (explained above my thoughts)